shared/tree-sitter-extractor/src/extractor/simple.rs

```diff
@@ -0,0 +1,155 @@
+use crate::trap;
+use rayon::prelude::*;
+use std::collections::HashMap;
+use std::ffi::{OsStr, OsString};
+use std::fs::File;
+use std::io::BufRead;
+use std::path::{Path, PathBuf};
+
+use crate::diagnostics;
+use crate::node_types;
+
+pub struct LanguageSpec {
+    pub prefix: &'static str,
+    pub ts_language: tree_sitter::Language,
+    pub node_types: &'static str,
+    pub file_extensions: Vec<OsString>,
+}
+
+pub struct Extractor {
+    pub prefix: String,
+    pub languages: Vec<LanguageSpec>,
+    pub trap_dir: PathBuf,
+    pub source_archive_dir: PathBuf,
+    pub file_list: PathBuf,
+}
+
+impl Extractor {
+    pub fn run(&self) -> std::io::Result<()> {
+        let diagnostics = diagnostics::DiagnosticLoggers::new(&self.prefix);
+        let mut main_thread_logger = diagnostics.logger();
+        let num_threads = match crate::options::num_threads() {
+            Ok(num) => num,
+            Err(e) => {
+                main_thread_logger.write(
+                    main_thread_logger
+                        .new_entry("configuration-error", "Configuration error")
+                        .message(
+                            "{}; defaulting to 1 thread.",
+                            &[diagnostics::MessageArg::Code(&e)],
+                        )
+                        .severity(diagnostics::Severity::Warning),
+                );
+                1
+            }
+        };
+        tracing::info!(
+            "Using {} {}",
+            num_threads,
+            if num_threads == 1 {
+                "thread"
+            } else {
+                "threads"
+            }
+        );
+        let trap_compression = match trap::Compression::from_env("CODEQL_QL_TRAP_COMPRESSION") {
```

Comment:
This environment variable should be language specific. `CODEQL_{LANGUAGE}_TRAP_COMPRESSION`